### PR TITLE
Update year and some dependencies

### PR DESCRIPTION
**Pull request by Cito**
- Update year of copyright
- Update and clean-up GitHub Actions
- Remove pytest-pofiling from dev dependencies (unmaintained, does not work properly with Python 3.12)
- Update some dependencies

### CODE REVIEW

File: .github/workflows/check_template_files.yaml [hunk 1]
```diff
@@ -5,13 +5,22 @@ on: push
...
+      - name: Checkout repository
+        id: checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Python 3.12
+        id: setup-python
+        uses: actions/setup-python@v5
         with:
-          python-version: 3.9
+          python-version: '3.12'
```
Comment (mephenor) [inline .github/workflows/check_template_files.yaml:20]: This is not in the other workflows. Why is this needed here?

**Review by mephenor — APPROVED** [review]

File: .github/workflows/check_template_files.yaml (hunk 1 above)
Comment (Cito) [reply .github/workflows/check_template_files.yaml:20]: Because we just setup Python here and do not execute the full commons action (which has a default Python version).

Comment (mephenor) [reply .github/workflows/check_template_files.yaml:20]: Missed tht detail, thanks.
Then this should be mergeable
